Library/Homebrew/linkage_checker.rb

```diff
@@ -39,7 +46,7 @@ def check_dylibs
             @system_dylibs << dylib
           rescue Errno::ENOENT
             next if harmless_broken_link?(dylib)
-            @broken_dylibs << dylib
+            @broken_dylibs[dylib_to_dep(dylib)] << dylib
```

Comment:
What if `dylib_to_dep(dylib)` is `nil`?

Reply:
it just works :) (`nil` can be used as a key in hashes)

Reply:
It feels very strange to do so?

Reply:
the result of `dylib_to_dep` is supposed to be the name of a Homebrew formula that provides that library. It seems reasonable for `dylib_to_dep` to return `nil` if missing library is not provided by Homebrew.

Reply:
is there a `.keg_name` method that we could apply to `dylib`?

Reply:
Use a separate list in that case, I'd say.

Reply:
> Use a separate list

Can you clarify?

Reply:
Don't use `@broken_dylibs` with a `nil` key; use a separate instance variable for `@broken_deps[dylib_to_dep(dylib)]` and `@broken_dylibs`

Reply:
OK, makes sense. So when `dylib_to_dep(dylib)` resolves to a non-empty string - populate `broken_dep`, otherwise - `broken_dylib`